### PR TITLE
Using FetchContent to replace the submodules

### PR DESCRIPTION
**Pull request by mkaring**

### CODE REVIEW

File: extern/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,31 @@
+include( FetchContent )
+
+FetchContent_Declare(
+    googletest
+    GIT_REPOSITORY https://github.com/google/googletest.git
+    GIT_TAG        release-1.10.0
```
Comment (vilarion) [inline extern/CMakeLists.txt:6]: I believe the tag used here should be v1.10.x to get bugfixes for 1.10.

File: CMakeLists.txt [hunk 2]
```diff
@@ -1,6 +1,7 @@
 cmake_minimum_required(VERSION 3.13)
```
Comment (vilarion) [inline CMakeLists.txt:1]: cmake output has cloned entries, doesn't seem entirely clean to me:
-- Boost version: 1.67.0
-- Boost version: 1.67.0
-- Found the following Boost libraries:
--   system
--   graph
-- Boost version: 1.67.0
-- Boost version: 1.67.0
-- Boost version: 1.67.0
-- Boost version: 1.67.0

Comment (vilarion) [inline CMakeLists.txt:1]: CMake Warning (dev) at /home/andreas/src/cmake-server/_deps/luabind-src/CMakeLists.txt:8 (project):
  Policy CMP0048 is not set: project() command manages VERSION variables.
  Run "cmake --help-policy CMP0048" for policy details.  Use the cmake_policy
  command to set the policy and suppress this warning.

  The following variable(s) would be set to empty:

    PROJECT_VERSION
    PROJECT_VERSION_MAJOR
    PROJECT_VERSION_MINOR
    PROJECT_VERSION_PATCH

**Review by vilarion — CHANGES_REQUESTED** [review]

File: CMakeLists.txt (hunk 2 above)
Comment (mkaring) [reply CMakeLists.txt:1]: This is resolved.

File: extern/CMakeLists.txt (hunk 1 above)
Comment (mkaring) [reply extern/CMakeLists.txt]: Changed as requested.

File: CMakeLists.txt (hunk 2 above)
Comment (mkaring) [reply CMakeLists.txt:1]: Did as  much as I could.

**Review by vilarion — APPROVED** [review]

**Pull request merged**